### PR TITLE
added crispritz recipe

### PR DESCRIPTION
**Pull request by samuelecancellieri**
:information_source:
If this is your first pull request to Bioconda please ping `@bioconda/core` so it can be reviewed and to request being added as a member of the Bioconda team. Members of the Bioconda team are able to merge their own pull requests once tests and linting have passed.

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

i send this pull request to merge my new recipe to the bioconda git.
thanks

### CODE REVIEW

**Comment by dpryan79** [comment]
Can you pull the `master` branch from upstream? The linting error you're running into is due to that. BTW, in the future, it's best to create a new branch in your fork to prevent merge issues.

**Comment by samuelecancellieri** [comment]
> Can you pull the `master` branch from upstream? The linting error you're running into is due to that. BTW, in the future, it's best to create a new branch in your fork to prevent merge issues.

hi, thanks for the help, i opened an issue because the recipe fail the macOS X test, as i understand the problem is the c++ compiler, but i need the specific compiler and the specific version to compile and run without errors, as i ask in the issue #15103 

Thanks a lot.

Samuele

**Comment by epruesse** [comment]
@biocondabot recheck

File: recipes/crispritz/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+make
+mkdir -p $PREFIX/bin
+chmod 700 -R *
+cp crispritz.py $PREFIX/bin
+cp buildTST $PREFIX/bin
+cp searchTST $PREFIX/bin
+cp searchBruteForce $PREFIX/bin
+cp -R sourceCode/Python_Scripts/ $PREFIX/bin
```
Comment (epruesse) [inline recipes/crispritz/build.sh]: That's not OK. Can't have random stuff in $PREFIX/bin. Needs to be installed to e.g. `$PREFIX/opt/crisptritz`. Try installing everything there and then creating symlinks in `$PREFIX/bin`.

Right now, we have in the package:
```
bin/buildTST
bin/searchTST
bin/searchBruteForce
bin/crispritz.py
bin/Python_Scripts/Plot/gecko/gecko.guide.txt
bin/Python_Scripts/Plot/gecko/gecko.CTCF.Count.txt
bin/Python_Scripts/Plot/gecko/gecko.Exons.Count.txt
bin/Python_Scripts/Plot/gecko/gecko.Promoters.Count.txt
bin/Python_Scripts/Plot/gecko/gecko.Introns.Count.txt
bin/Python_Scripts/Plot/gecko/gecko.DNAse.Count.txt
bin/Python_Scripts/Plot/gecko/gecko.reference.profile.xls
bin/Python_Scripts/Enrichment/bcf_query.sh
bin/Python_Scripts/Enrichment/enricher.py
bin/Python_Scripts/Annotator/annotator.py
bin/Python_Scripts/Plot/radar_chart.py
```

Comment (samuelecancellieri) [reply recipes/crispritz/build.sh]: Hi,

if I understand correctly, I need to move all the file and scripts in another directory different from bin.

But the script I use in the program shouldn't be located in the bin directory?? For example crispritz.py is the wrapper to all the other scripts, like the three C++ executable (buildTST, searchTST, searchBruteForce) and all the other python scripts (enricher.py, annotator.py, radar_chart.py). I should move all the files in a secondary directory?

Thanks.

Samuele

Comment (dpryan79) [reply recipes/crispritz/build.sh]: Just put the `Python_Scripts` directory elsewhere and symlink whatever is needed to `bin`.

Comment (samuelecancellieri) [reply recipes/crispritz/build.sh]: All the software function are called by crispritz.py, the main wrapper.
The c++ executable are saved in the $PREFIX/bin and the wrapper assume that this is the correct location, all the other python scripts and the bcf_query bash script, are located in the respective directory, the plot directory also contain text file using during the execution of the python code. The main wrapper (crispritz.py) assumes the all the files are in $PREFIX/bin or in $PREFIX/bin/Python_Scripts

File: recipes/crispritz/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  url: https://github.com/InfOmics/CRISPRitz/archive/master.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True # [osx]
+
+requirements:
+  build:
+    - {{compiler('cxx')}} =5.4
```
Comment (dpryan79) [inline recipes/crispritz/meta.yaml]: Please get rid of `=5.4`, that's likely to break other packages in the same env.

File: recipes/crispritz/build.sh (hunk 1 above)
Comment (dpryan79) [reply recipes/crispritz/build.sh]: OK, if the software assumes that structure there's nothing that can be changed.

Comment (dpryan79) [reply recipes/crispritz/build.sh]: Actually, then you can probably do what Elmar mentioned and just put everything under `opt` or `share` and then create symlinks to `bin`.

Comment (samuelecancellieri) [reply recipes/crispritz/build.sh]: I modified the crispritz.py file, now should check for the scripts in opt/crispritz

**Comment by epruesse** [comment]
Since this is self publishing, might it be easier to fix the tool itself? Add a setup.py etc?

Also, please tag a release. The URL you have there probably always points to master, so it’s not stable.

**Comment by samuelecancellieri** [comment]
> Since this is self publishing, might it be easier to fix the tool itself? Add a setup.py etc?
> 
> Also, please tag a release. The URL you have there probably always points to master, so it’s not stable.

It's OK to adjust the wrapper, I don't want to create more installation files that are more difficult to maintain.
I also set the tag version and modified the link to download from GitHub.

File: recipes/crispritz/build.sh (hunk 1 above)
Comment (epruesse) [inline recipes/crispritz/build.sh]: Still need to move those things somewhere else. :)

Comment (samuelecancellieri) [reply recipes/crispritz/build.sh]: Sorry I forgot to change the file, now it's correct.

Comment (samuelecancellieri) [reply recipes/crispritz/build.sh]: > Still need to move those things somewhere else. :)

Hi, I updated the recipe and the build file.

Thanks.

**Comment by samuelecancellieri** [comment]
@bioconda/core

**Comment by dpryan79** [comment]
You pinged? Looks like this got merged.

**Comment by samuelecancellieri** [comment]
> You pinged? Looks like this got merged.

Yes, sorry. I was reading the one-time setup, and I pinged to ask if it will be possible to be added as bioconda team member.
Thanks.

**Comment by dpryan79** [comment]
Sure thing!

**Comment by samuelecancellieri** [comment]
> Sure thing!

Wonderful.
Thank you.

**Pull request merged**